src/unity/python/turicreate/toolkits/drawing_recognition/drawing_recognition.py

```diff
@@ -0,0 +1,480 @@
+    def _native_name(cls):
+        return "drawing_recognition"
+
+    def _get_native_state(self):
+        state = self.__proxy__.get_state()
+        mxnet_params = state['model'].collect_params()
+        state['model'] = _mxnet_utils.get_gluon_net_params_state(mxnet_params)
+        return state
+
+    def _get_version(self):
+        return 1
+
+    @classmethod
+    def _load_version(cls, state, version):
+        _tkutl._model_version_check(version, 1)
+        net = Model(num_classes = len(state['classes']), prefix = 'model0_')
+        ctx = _mxnet_utils.get_mxnet_context(max_devices=state['batch_size'])
+        net_params = net.collect_params()
+        _mxnet_utils.load_net_params_from_state(
+            net_params, state['model'], ctx=ctx 
+            )
+        state['model'] = net
+        return DrawingRecognition(state)
+
+    def export_coreml(self, filename, verbose=False):
+        import mxnet as _mx
+        from .._mxnet_to_coreml import _mxnet_converter
+        import coremltools
+        from coremltools.models import datatypes, neural_network
+
+        batch_size = 1
+        image_shape = (batch_size,) + (1, BITMAP_WIDTH, BITMAP_HEIGHT)
+        s_image = _mx.sym.Variable('bitmap',
+            shape=image_shape, dtype=_np.float32)
+
+        net = _copy(self.model)
+        s_ymap = net(s_image)
+        
+        mod = _mx.mod.Module(symbol=s_ymap, label_names=None, data_names=['bitmap'])
+        mod.bind(for_training=False, data_shapes=[('bitmap', image_shape)])
+        mod.init_params()
+        
+        arg_params, aux_params = mod.get_params()
+        net_params = net.collect_params()
+
+        new_arg_params = {}
+        for k, param in arg_params.items():
+            new_arg_params[k] = net_params[k].data(net_params[k].list_ctx()[0])
+        new_aux_params = {}
+        for k, param in aux_params.items():
+            new_aux_params[k] = net_params[k].data(net_params[k].list_ctx()[0])
+        mod.set_params(new_arg_params, new_aux_params)
+
+        coreml_model = _mxnet_converter.convert(mod, mode='classifier',
+                                class_labels=self.classes,
+                                input_shape=[('bitmap', image_shape)],
+                                builder=None, verbose=verbose,
+                                preprocessor_args={'image_input_names':['bitmap']},
+                                is_drawing_recognition=True)
+
+        from turicreate.toolkits import _coreml_utils
+        model_type = "drawing classifier"
+        coreml_model.short_description = _coreml_utils._mlmodel_short_description(model_type)
+        coreml_model.input_description['bitmap'] = u'bitmap'
+        coreml_model.output_description['probabilities'] = 'Prediction probabilities'
+        coreml_model.output_description['classLabel'] = 'Class Label of Top Prediction'
+        coreml_model.save(filename)
+        return coreml_model
+
+    def _predict_with_options(self, input_dataset):
```

Comment:
What are the options? Why do we need a separate function here?

Reply:
A better name for this would be `predict_with_probabilities`. The code here is shared by both `evaluate` and `predict`. While `predict` only cares about the label column in the SFrame that's returned, `evaluate` cares about the `p robabilities` column as well (to compute metrics like AUC). I'll let @fareeha chime in if there's something I'm missing.